File: Python/Problems/Problems/problem3779.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem3779(Problem):
    def test(self) -> bool:
        nums = [3, 8, 3, 6, 5, 8]

        expected = 1

        result = self.minOperations(nums)

        print(result)

        return result == expected

    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        present = set()
        for i in range(n - 1, -1, -1):
            if nums[i] in present:
                return (i + 1 + 2) // 3
            present.add(nums[i])

        return 0

File: C++/Problems/Problems/problem_3779.cpp
#include "problem_3779.h"

#include <unordered_set>

bool problem_3779::test() {
    std::vector nums { 3, 8, 3, 6, 5, 8 };

    int expected = 1;

    auto result = minOperations(nums);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_3779::minOperations(std::vector<int> &nums) {
    int n = nums.size();
    std::unordered_set<int> present;
    for (int i = n - 1; i >= 0; --i)
    {
        if (present.contains(nums[i]))
            return (i + 1 + 2) / 3;
        present.insert(nums[i]);
    }

    return 0;
}
